addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/channel/TypeColor.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.knx.internal.channel;
+
+import static java.util.stream.Collectors.toSet;
+import static org.openhab.binding.knx.KNXBindingConstants.*;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+import tuwien.auto.calimero.dptxlator.DPTXlator3BitControlled;
+import tuwien.auto.calimero.dptxlator.DPTXlator8BitUnsigned;
+import tuwien.auto.calimero.dptxlator.DPTXlatorBoolean;
+import tuwien.auto.calimero.dptxlator.DPTXlatorRGB;
+
+/**
+ * color channel type description
+ *
+ * @author Simon Kaufmann - initial contribution and API.
+ * @author Helmut Lehmeyer
+ *
+ */
+@NonNullByDefault
+class TypeColor extends KNXChannelType {
+
+    TypeColor() {
+        super(CHANNEL_COLOR, CHANNEL_COLOR_CONTROL);
+    }
+
+    @Override
+    protected Set<String> getAllGAKeys() {
+        return Stream.of(SWITCH_GA, POSITION_GA, INCREASE_DECREASE_GA, HSB_GA).collect(toSet());
+    }
+
+    @Override
+    protected String getDefaultDPT(String gaConfigKey) {
+        if (Objects.equals(gaConfigKey, HSB_GA)) {
```

Comment:
Just a side-note: since HSB_GA is a non-null constant and gaConfigKey non-null by annotation, why don't you use `gaConfigKey.equals(HSB_GA)`?

Reply:
Done!